src/overlays/actors/ovl_Boss_Dodongo/z_boss_dodongo.c

```diff
@@ -51,311 +44,1717 @@ const ActorInit Boss_Dodongo_InitVars = {
-        },
-        { 45, { { 0, 0, 0 }, 30 }, 100 },
-    },
-    {
-        {
-            ELEMTYPE_UNK2,
-            { 0xFFCFFFFF, 0x00, 0x04 },
-            { 0xFFCFFFFF, 0x00, 0x00 },
-            TOUCH_ON | TOUCH_SFX_NORMAL,
-            BUMP_ON,
-            OCELEM_ON,
-        },
-        { 46, { { 0, 0, 0 }, 30 }, 100 },
-    },
-    {
-        {
-            ELEMTYPE_UNK2,
-            { 0xFFCFFFFF, 0x00, 0x04 },
-            { 0xFFCFFFFF, 0x00, 0x00 },
-            TOUCH_ON | TOUCH_SFX_NORMAL,
-            BUMP_ON,
-            OCELEM_ON,
-        },
-        { 47, { { 0, 0, 0 }, 40 }, 100 },
-    },
+#include "z_boss_dodongo_data.c"
+
+static InitChainEntry sInitChain[] = {
+    ICHAIN_U8(targetMode, 5, ICHAIN_CONTINUE),
+    ICHAIN_S8(naviEnemyId, 12, ICHAIN_CONTINUE),
+    ICHAIN_F32_DIV1000(gravity, -3000.0f, ICHAIN_CONTINUE),
+    ICHAIN_F32(targetArrowOffset, 8200.0f, ICHAIN_STOP),
 };
 
-static ColliderJntSphInit D_808C73AC = {
-    {
-        COLTYPE_HIT3,
-        AT_ON | AT_TYPE_ENEMY,
-        AC_ON | AC_TYPE_PLAYER,
-        OC1_ON | OC1_TYPE_PLAYER,
-        OC2_TYPE_1,
-        COLSHAPE_JNTSPH,
-    },
-    19,
-    D_808C7100,
+extern SkeletonHeader D_0601B310;
+extern AnimationHeader D_0600F0D8;
+extern u64 D_030021D8[];
+extern AnimationHeader D_06008EEC;
+extern u64* D_06017410; // Title card
+extern AnimationHeader D_0601CAE0;
+extern AnimationHeader D_06008EEC;
+extern AnimationHeader D_060061D4;
+extern AnimationHeader D_0600DF38;
+extern AnimationHeader D_0600E848;
+extern AnimationHeader D_06004E0C;
+extern AnimationHeader D_060042A8;
+extern AnimationHeader D_06009D10;
+extern AnimationHeader D_0601D934;
+extern AnimationHeader D_06002D0C;
+extern AnimationHeader D_060042A8;
+extern AnimationHeader D_06001074;
+extern AnimationHeader D_06003CF8;
+extern s16 D_06015890[];
+extern s16 D_06017210[];
+extern s16 D_06015D90[];
+extern s16 D_06016390[];
+extern s16 D_06016590[];
+extern s16 D_06016790[];
+extern s16 D_06015990[];
+extern s16 D_06015F90[];
+extern s16 D_06016990[];
+extern s16 D_06016E10[];
+extern Gfx* D_06009D50[];
+extern Gfx* D_06009DD0[];
+
+void func_808C1190(s16* arg0, u8* arg1, s16 arg2) {
+    if (arg2[arg1] != 0) {
+        arg0[arg2 / 2] = 0;
+    }
+}
+
+void func_808C11D0(s16* arg0, u8* arg1, s16 arg2) {
+    if (arg1[arg2] != 0) {
+        arg0[arg2] = 0;
+    }
+}
+
+void func_808C1200(s16* arg0, u8* arg1, s16 arg2) {
+    if (arg1[arg2] != 0) {
+        arg0[arg2] = 0;
+    }
+}
+
+void func_808C1230(s16* arg0, u8* arg1, s16 arg2) {
+    s16 index;
+
+    if (arg1[arg2] != 0) {
+        index = ((arg2 & 0xF) + ((arg2 & 0xF0) * 2));
+        arg0[index + 16] = 0;
+        arg0[index] = 0;
+    }
+}
+
+void func_808C1278(s16* arg0, u8* arg1, s16 arg2) {
+    s16 index;
+
+    if (arg1[arg2] != 0) {
+        index = ((arg2 & 0xF) * 2) + ((arg2 & 0xF0) * 2);
+        arg0[index + 1] = 0;
+        arg0[index] = 0;
+    }
+}
+
+void func_808C12C4(void* arg1, s16 arg2) {
```

Comment:
This seems to be used as a `u8*` by all functions called so I suggest using that type in this function as well. Also the args aren't numbered as you usually do so I think that should be fixed for consistency.
```suggestion
void func_808C12C4(u8* arg0, s16 arg1) {
```